docs/lab-hooks.md

```diff
@@ -0,0 +1,352 @@
+# Lab Hooks
+
+This document provides a summary of all the lab (aka assessment) hooks available to an instructor.
+
+Lab hooks are defined in the lab's configuration file, `<labname>.rb`. The configuration file is located in the lab's directory, at the path `<coursename>/<labname>/<labname>.rb`.
+
+To make changes live, you must select the "Reload config file" option on the lab's index page. You can also upload a new config file from the lab's setting page.
+
+To debug the hooks, you can make use of the `ASSESSMENT_LOGGER.log(<expr>)` method to print output into the lab's `log.txt` file.
+
+!!! Danger "Function Arity"
+    When defining the hooks below, be sure that they take the correct number of arguments.
+    Failure to do so might leave your assessment in a hard-to-recover state. 
+
+## Modify Submission Score
+
+Hook: `modifySubmissionScores`
+
+By default, the scores output by the autograder will be directly assigned to the individual problem scores. This hook allows you to override the score calculation for a lab.
```

Comment:
outputted seems more apt here

Reply:
![Screenshot 2024-04-01 at 16.35.03.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/mdqiNyhEZukqZ5c333FS/b72a496b-45c6-4273-97ad-87415ba571e1.png)

output seems to be the more common word, so I'll leave it as-is